app/src/lib/stores/app-store.ts

```diff
@@ -962,9 +971,49 @@ export class AppStore extends TypedBaseStore<IAppState> {
       this.currentAheadBehindUpdater.insert(from, to, aheadBehind)
     }
 
-    this.updateOrSelectFirstCommit(repository, commitSHAs)
+    if (!enableMergeConflictDetection()) {
+      this.updateOrSelectFirstCommit(repository, commitSHAs)
+      return this.emitUpdate()
+    } else {
+      this.updateCompareState(repository, () => ({
+        mergeStatus: { kind: MergeResultKind.Loading },
+      }))
+
+      this.emitUpdate()
+
+      this.updateOrSelectFirstCommit(repository, commitSHAs)
+
+      if (this.currentMergeTreePromise != null) {
+        return this.currentMergeTreePromise
+      }
 
-    return this.emitUpdate()
+      if (tip.kind === TipState.Valid) {
+        const mergeTreePromise = promiseWithMinimumTimeout(
+          () => mergeTree(repository, tip.branch, action.branch),
+          500
+        ).then(mergeStatus => {
+          this.updateCompareState(repository, () => ({
+            mergeStatus,
+          }))
+
+          this.emitUpdate()
+        })
+
+        const cleanup = () => (this.currentMergeTreePromise = null)
+
+        mergeTreePromise.then(cleanup, cleanup)
```

Comment:
@iAmWillShepherd `.finally` isn't valid to attach to a callback - are you referring to `try { } finally { }` that would require `await`-ing the promise?

Reply:
We discussed this in slack. Basically, we can't use this in our project yet because we are on an older version of node. https://github.com/tc39/proposal-promise-finally

Reply:
We can start using this in TS when we target `es2018` or later, but it's also only available on Node 10 or later, so we'd need to wait for Electron to start shipping Node 10 too.